src/main/c/ssl.c

```diff
@@ -1201,6 +1202,11 @@ TCN_IMPLEMENT_CALL(jlong /* SSL * */, SSL, newSSL)(TCN_STDARGS,
         tcn_ThrowException(e, "cannot create new ssl");
         return 0;
     }
+
+    // Store the handshakeCount in the SSL instance.
+    *handshakeCount = 0;
```

Comment:
nit: do we need to check for a memory allocation failure....we would be pretty hosed at that point anyways but WDYT?